src/components/PasswordChange/PasswordChange.js

```diff
@@ -1,68 +1,50 @@
-import React from 'react';
+import React, {useState} from 'react';
 import PropTypes from 'prop-types';
 
+import {useFormInput} from '../../hooks/hooks';
 import {withFirebase} from '../Firebase';
 
-const INITIAL_STATE = {
-  passwordOne: '',
-  passwordTwo: '',
-  error: null
-};
-
-class PasswordChangeForm extends React.Component {
-  constructor(props) {
-    super(props);
-
-    this.state = {...INITIAL_STATE};
-  }
+const PasswordChangeForm = props => {
+  const passwordOne = useFormInput('');
+  const passwordTwo = useFormInput('');
+  const [error, setError] = useState(null);
 
-  onSubmit = e => {
-    const {passwordOne} = this.state;
-
-    this.props.firebase
-      .doPasswordUpdate(passwordOne)
-      .then(() => {
-        this.setState({...INITIAL_STATE});
-      })
+  const onSubmit = e => {
+    props.firebase
+      .doPasswordUpdate(passwordOne.value)
+      .then(() => useFormInput)
       .catch(error => {
-        this.setState({error});
+        setError(error);
       });
 
     e.preventDefault();
   };
 
-  onChange = e => {
-    this.setState({[e.target.name]: e.target.value});
-  };
-
-  render() {
-    const {passwordOne, passwordTwo} = this.state;
-
-    const isInvalid = passwordOne !== passwordTwo || passwordOne === '';
-
-    return (
-      <form onSubmit={this.onSubmit}>
-        <input
-          name="passwordOne"
-          value={passwordOne}
-          onChange={this.onChange}
-          type="password"
-          placeholder="New password"
-        />
-        <input
-          name="passwordTwo"
-          value={passwordTwo}
-          onChange={this.onChange}
-          type="password"
-          placeholder="Confirm new password"
-        />
-        <button disabled={isInvalid} type="submit">
-          Reset my password
-        </button>
-      </form>
-    );
-  }
-}
+  const isInvalid =
+    passwordOne.value !== passwordTwo.value || passwordOne.value === '';
+
+  return (
+    <form onSubmit={onSubmit}>
+      <input
+        name="passwordOne"
+        type="password"
+        placeholder="New password"
+        {...passwordOne}
```

Comment:
I see. that's a clever way. But why should it be a hook?

Reply:
It is a recurring pattern and saves writing code.
I can explain it to you the next time that we meet.